src/main/java/com/browserstack/automate/ci/common/tracking/PluginsTracker.java

```diff
@@ -27,14 +27,42 @@ public PluginsTracker(final String username, final String accessKey) {
         this.username = username;
         this.accessKey = accessKey;
         this.trackingId = Tools.getUniqueString(true, true);
+        initializeClient();
     }
 
     public PluginsTracker() {
         this.username = null;
         this.accessKey = null;
         this.trackingId = Tools.getUniqueString(true, true);
+        initializeClient();
     }
-
+    
+    public void initializeClient(){
+
+        String username = JenkinsProxySettings.getUsername();
+        String password = JenkinsProxySettings.getPassword();
+        Authenticator proxyAuthenticator = new Authenticator() {
```

Comment:
we wouldn't need the `proxyAuthenticator` if `username` and `password` are null right?...

Reply:
Done